targets/TARGET_Cypress/scripts/mbed_set_post_build_cypress.cmake

```diff
@@ -3,48 +3,56 @@
 
 include(mbed_set_post_build)
 
-set(MBED_POST_BUILD_CYPRESS_DIR "${CMAKE_CURRENT_LIST_DIR}")
-
 #
 # Merge Cortex-M4 HEX and a Cortex-M0 HEX.
 #
-function(mbed_post_build_psoc6_merge_hex mbed_target_name)
-    find_package(Python3)
+macro(mbed_post_build_psoc6_merge_hex mbed_target_name)
+    if("${mbed_target_name}" STREQUAL "${MBED_TARGET}")
+        function(mbed_post_build_function target)
+            find_package(Python3)
 
-    # Copy ${ARGN} to a variable first as it cannot be used directly with
-    # the list() command
-    set (extra_macro_args ${ARGN})
+            # Copy ${ARGN} to a variable first as it cannot be used directly with
+            # the list() command
+            set (extra_macro_args ${ARGN})
 
-    # Get the number of arguments past the last expected argument
-    list(LENGTH extra_macro_args num_extra_args)
+            # Get the number of arguments past the last expected argument
+            list(LENGTH extra_macro_args num_extra_args)
 
-    if(${num_extra_args} GREATER 0)
-        # Get extra argument as `cortex_m0_hex`
-        list(GET extra_macro_args 0 cortex_m0_hex)
-        set(post_build_command
-            COMMAND ${Python3_EXECUTABLE} ${MBED_POST_BUILD_CYPRESS_DIR}/PSOC6.py
-                merge
-                --elf ${CMAKE_BINARY_DIR}/$<TARGET_PROPERTY:mbed-post-build-bin-${mbed_target_name},application>.elf
-                --m4hex ${CMAKE_BINARY_DIR}/$<TARGET_PROPERTY:mbed-post-build-bin-${mbed_target_name},application>.hex
-                --m0hex ${cortex_m0_hex}
-        )
-    else()
-        set(post_build_command
-            COMMAND ${Python3_EXECUTABLE} ${MBED_POST_BUILD_CYPRESS_DIR}/PSOC6.py
-                merge
-                --elf ${CMAKE_BINARY_DIR}/$<TARGET_PROPERTY:mbed-post-build-bin-${mbed_target_name},application>.elf
-                --m4hex ${CMAKE_BINARY_DIR}/$<TARGET_PROPERTY:mbed-post-build-bin-${mbed_target_name},application>.hex
-        )
-    endif()
+            if(${num_extra_args} GREATER 0)
```

Comment:
Preexisting but this seems like a super-weak check for "cortex_m0_hex". Almost "programming by coincidence". There's some tacit knowledge here that should be made explicit.

Reply:
It looks like `$[ARGN}` is the standard way to support optional arguments. If checking `${num_extra_args} GREATER 0` is too rudimentary, CMake has a built-in `cmake_parse_argument` which we use for greentea: https://github.com/ARMmbed/mbed-os/blob/0f5e062b69843a112f9e3d5eb672bd01d0b836a3/tools/cmake/mbed_greentea.cmake#L26-L39

Is this what you would suggest?

Reply:
That is what I would suggest.